executor/index_merge_reader.go

```diff
@@ -481,13 +494,14 @@ func (e *IndexMergeReaderExecutor) handleHandlesFetcherPanic(ctx context.Context
 
 // Close implements Exec Close interface.
 func (e *IndexMergeReaderExecutor) Close() error {
-	if e.finished == nil {
+	if !e.workerStarted || e.finished == nil {
```

Comment:
1. `IndexMergeProcessWorker` is likely to be started already even if `e.workerStarted` is false. Thus we should not `return nil` when `e.workerStarted` is false.
2. We need to `checkGoroutineExists` in `TestIndexMergeReaderClose` for `IndexMergeProcessWorker`.